00-framework-01-maturity.md

```diff
@@ -0,0 +1,155 @@
+# libp2p specification framework – the five-stage maturity process
+
+> Author: @raulk
+
+> Revision: r0, 2019-05-21
+
+## Prelude
+
+This document establishes a framework to foster rapid and incremental libp2p
+specification development, aiming to reduce the barrier to entry for new
+ideas, increasing the throughput of breakthrough creative proposals, promoting
+their evolution and adoption within the community, while maximising consensus
+around a common lifecycle and criteria for progression across stages.
+
+We propose a five-stage process to classify the maturity of a specification in
+terms of completeness, demonstrability of implementation, community
+acceptance, and level of technical detail. It is somewhat inspired in the W3C
+Process [0].
+
+  1. `Working Draft`
+  2. `Candidate Recommendation`
+  3. `Recommendation`
+  4. `Deprecated`
+  5. `Abandoned Draft`
+
+## Stage 1: Working draft
```

Comment:
I agree with @whyrusleeping, let's not call it stages. Otherwise this would imply progression from 1->5.

Reply:
I would go with what the w3c process uses, `Levels`.

Reply:
See https://github.com/libp2p/specs/pull/169#issuecomment-494741011.